src/preamble.js

```diff
@@ -797,6 +801,125 @@ function resetPrototype(constructor, attrs) {
 }
 #endif
 
+#if WASM_ASYNC_COMPILATION
+function instantiateArrayBuffer(binaryFile, imports, receiver) {
```

Comment:
This large function looks like it's been moved around, which makes the diff harder to read. Was moving it necessary, or just for organization? (if the latter then perhaps an initial NFC PR?)

Reply:
It was previously in the `createWasm` closure and used some of the variables in the closure. I needed to called it outside of `createWasm`.  I could leave it in there and expose it, but it's would no longer need to be dependant on the variables.

Reply:
I see, thanks. I manually inspected the diff now and it's just those variable changes, indeed.